File: 2001-3000/2068-check-whether-two-strings-are-almost-equivalent/2068-check-whether-two-strings-are-almost-equivalent.cpp
class Solution {
public:
    bool checkAlmostEquivalent(string word1, string word2) {
        int n=word1.size();
        int m=word2.size();

       int hash1[26]={0};
       int hash2[26]={0};
       for(int i=0;i<n;i++){
            hash1[word1[i]-'a']+=1;
       }
       for(int i=0;i<m;i++){
        hash2[word2[i]-'a']+=1;
       }

       for(int i=0;i<n;i++){
        if(abs(hash1[word1[i]-'a']-hash2[word1[i]-'a'])>3) return false;
       }
        for(int i=0;i<m;i++){
        if(abs(hash1[word2[i]-'a']-hash2[word2[i]-'a'])>3) return false;
       }
       return true;
    }
};

File: 2001-3000/2068-check-whether-two-strings-are-almost-equivalent/2068-check-whether-two-strings-are-almost-equivalent.java
class Solution {
    public boolean checkAlmostEquivalent(String word1, String word2) {
        
        int[] freq = new int[26];

        // Count characters of word1
        for (char ch : word1.toCharArray()) {
            freq[ch - 'a']++;
        }

        // Subtract characters of word2
        for (char ch : word2.toCharArray()) {
            freq[ch - 'a']--;
        }

        // Check difference
        for (int i = 0; i < 26; i++) {
            if (Math.abs(freq[i]) > 3) {
                return false;
            }
        }

        return true;
    }
}